### PR TITLE
Moving to SDK 2.0

### PR DESCRIPTION
**Pull request by serkantkaraca**
Moving client to .Net Standard SDK 2.0

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [x] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [x] If applicable, the public code is properly documented.
- [x] Pull request includes test coverage for the included changes.
- [x] The code builds without any errors.

### CODE REVIEW

File: src/Microsoft.Azure.EventHubs.Processor/Microsoft.Azure.EventHubs.Processor.csproj [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <AssemblyTitle>Microsoft.Azure.EventHubs.Processor</AssemblyTitle>
     <VersionPrefix>1.0.3</VersionPrefix>
     <Authors>Microsoft</Authors>
-    <TargetFrameworks>net451;netstandard1.3;uap10.0;</TargetFrameworks>
+    <TargetFrameworks>net451;netstandard2.0;uap10.0;</TargetFrameworks>
```
Comment (nemakam) [inline src/Microsoft.Azure.EventHubs.Processor/Microsoft.Azure.EventHubs.Processor.csproj]: >net451 [](start = 22, length = 6)

Do you want to move this to 461 as well? [https://github.com/dotnet/standard/blob/master/docs/versions.md](https://github.com/dotnet/standard/blob/master/docs/versions.md)

File: src/Microsoft.Azure.EventHubs/Microsoft.Azure.EventHubs.csproj [hunk 2]
```diff
@@ -5,7 +5,7 @@
     <AssemblyTitle>Microsoft.Azure.EventHubs</AssemblyTitle>
     <VersionPrefix>1.0.3</VersionPrefix>
```
Comment (nemakam) [inline src/Microsoft.Azure.EventHubs/Microsoft.Azure.EventHubs.csproj]: ><VersionPrefix>1.0.3</VersionPrefix> [](start = 4, length = 36)

considering this is a breaking change (raising the target framework), update the versionPrefix to 2.0

**Comment by nemakam** [comment]
>     <Version>1.0.3</Version>

You might not need versionPrefix and version at the same time. Removing it might help with removing redundancy

---
Refers to: src/Microsoft.Azure.EventHubs/Microsoft.Azure.EventHubs.csproj:28 in a74b926. [](commit_id = a74b92694cceb0f75949d10fb6450982f8c5074c, deletion_comment = False)

**Comment by nemakam** [comment]
>   <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard1.3' ">

These need to be updated.
Same for all the itemGroup conditions in all csproj

---
Refers to: src/Microsoft.Azure.EventHubs/Microsoft.Azure.EventHubs.csproj:46 in a74b926. [](commit_id = a74b92694cceb0f75949d10fb6450982f8c5074c, deletion_comment = False)

File: src/Microsoft.Azure.EventHubs.Processor/Microsoft.Azure.EventHubs.Processor.csproj (hunk 1 above)
Comment (serkantkaraca) [reply src/Microsoft.Azure.EventHubs.Processor/Microsoft.Azure.EventHubs.Processor.csproj]: Done

File: src/Microsoft.Azure.EventHubs/Microsoft.Azure.EventHubs.csproj (hunk 2 above)
Comment (serkantkaraca) [reply src/Microsoft.Azure.EventHubs/Microsoft.Azure.EventHubs.csproj]: Done

**Comment by nemakam** [comment]
>     <PackageTargetFallback Condition=" '$(TargetFramework)' == 'netcoreapp1.0' ">$(PackageTargetFallback);dnxcore50;portable-net46+win8</PackageTargetFallback>

Update this as well

---
Refers to: test/Microsoft.Azure.EventHubs.Tests/Microsoft.Azure.EventHubs.Tests.csproj:12 in 1a5ae4b. [](commit_id = 1a5ae4beb8e99034afd8c9d6cdf135209f65b899, deletion_comment = False)

**Comment by nemakam** [comment]
>     <GenerateAssemblyConfigurationAttribute>false</GenerateAssemblyConfigurationAttribute>

This is mostly not required with 2.0 sicne runtime framework is same as the target version. 

---
Refers to: test/Microsoft.Azure.EventHubs.Tests/Microsoft.Azure.EventHubs.Tests.csproj:14 in 1a5ae4b. [](commit_id = 1a5ae4beb8e99034afd8c9d6cdf135209f65b899, deletion_comment = False)

**Comment by nemakam** [comment]
>     - DotNetRunTime: net46

461 :)

---
Refers to: appveyor.yml:23 in 1b8ae6c. [](commit_id = 1b8ae6c05265066afbb5c44a15b7459fd458af1b, deletion_comment = False)

**Comment by nemakam** [comment]
> version: 1.0.{build}

2.0 ?

---
Refers to: appveyor.yml:1 in 1b8ae6c. [](commit_id = 1b8ae6c05265066afbb5c44a15b7459fd458af1b, deletion_comment = False)

**Comment by nemakam** [comment]
>   <ItemGroup Condition=" '$(TargetFramework)' == 'net451' ">

Missed this one?

---
Refers to: src/Microsoft.Azure.EventHubs/Microsoft.Azure.EventHubs.csproj:43 in 1b8ae6c. [](commit_id = 1b8ae6c05265066afbb5c44a15b7459fd458af1b, deletion_comment = False)

File: test/Microsoft.Azure.EventHubs.Tests/Microsoft.Azure.EventHubs.Tests.csproj [hunk 3]
```diff
@@ -1,26 +1,24 @@
...
     <GenerateAssemblyProductAttribute>false</GenerateAssemblyProductAttribute>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0" />
     <PackageReference Include="System.Net.WebSockets.Client" Version="4.0.0" />
     <PackageReference Include="xunit.runner.visualstudio" Version="2.2.0" />
-    <PackageReference Include="Microsoft.NETCore.Platforms" Version="1.1.0" />
+    <PackageReference Include="Microsoft.NETCore.Platforms" Version="2.0.0" />
```
Comment (nemakam) [inline test/Microsoft.Azure.EventHubs.Tests/Microsoft.Azure.EventHubs.Tests.csproj]: > <PackageReference Include="Microsoft.NETCore.Platforms" Version="2.0.0" /> [](start = 3, length = 75)

You can actually remove this. This is already present in the SDK that gets automatically imported

**Pull request merged**